cctools: Enforce xcode variant on Xcode 9+ systems

cjones051073 opened the pull request:
#### Description

The cctools port was recently updated to by default on systems using Xcode 9 or newer install using a new Xcode variant that simply installs wrapper scripts around the Xcode provided utilities. This was done to solve issues with the fact that previously the tool kit installed by cctools was based on Xcode 8, and this was starting to cause issues on Xcode9 and newer based systems.

One issue though was the fact the port previously already had a default on this systems, +llvm50. This means on regular updates this variant is preserved, instead of using the new Xcode variant. This has started to cause issues, specifically with gfortran from the gcc ports

https://trac.macports.org/ticket/56938

https://trac.macports.org/ticket/56938

My guess is this is due to the fact when built on the buildbots, the cctools port has the defaults, +xcode, whereas the user does not.

This update forces the use of this variant, by removing the previous one.

```
Oberon ~/Projects/MacPorts/ports > sudo port outdated
The following installed ports are outdated:
cctools                        895_5 < 895_6             
Oberon ~/Projects/MacPorts/ports > sudo port installed cctools
The following ports are currently installed:
  cctools @895_5+llvm50 (active)
Oberon ~/Projects/MacPorts/ports > sudo port upgrade outdated
Forcibly disabling variant llvm50
--->  Fetching archive for cctools
--->  Attempting to fetch cctools-895_6+xcode.darwin_17.noarch.tbz2 from http://mse.uk.packages.macports.org/sites/packages.macports.org/cctools
--->  Attempting to fetch cctools-895_6+xcode.darwin_17.noarch.tbz2 from http://lil.fr.packages.macports.org/cctools
--->  Attempting to fetch cctools-895_6+xcode.darwin_17.noarch.tbz2 from https://packages.macports.org/cctools
--->  Fetching distfiles for cctools
--->  Verifying checksums for cctools
--->  Extracting cctools
--->  Configuring cctools
--->  Building cctools
--->  Staging cctools into destroot
--->  Installing cctools @895_6+xcode
--->  Cleaning cctools
--->  Deactivating cctools @895_5+llvm50
--->  Cleaning cctools
--->  Activating cctools @895_6+xcode
--->  Cleaning cctools
--->  Scanning binaries for linking errors
--->  No broken files found.
--->  No broken ports found.
```

- [x] bugfix

###### Tested on

macOS 10.13.6 17G65
Xcode 9.4.1 9F2000 

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

ryandesign commented:
> My guess is this is due to the fact when built on the buildbots, the cctools port has the defaults, +xcode, whereas the user does not.

The buildbot always does installs, never upgrades.

cjones051073 commented:
Thanks. Any objections to fast forwarding merging this ? It addresses issues users are facing so I would like to get it out asap, ideally faster than the 3 day maintainer timeout (as its a fix to a previous update I would make the case its OK here)?

pmetzger commented:
I have no objection.
